File: frontend/src/lib/avatars.ts
/**
 * 16 hand-picked DiceBear avatars.
 * ids 0-7  → "Glow" group (adventurer style, warm tones)
 * ids 8-15 → "Cool" group (adventurer-neutral style, cool tones)
 *
 * Users pick the avatar that feels like them - this IS their identity.
 */

export interface AvatarDef {
  id: number;
  seed: string;
  style: "adventurer" | "adventurer-neutral";
  bg: string;   // hex without #
}

const A = "adventurer" as const;
const B = "adventurer-neutral" as const;

export const AVATARS: AvatarDef[] = [
  { id: 0,  seed: "Lily",  style: A, bg: "ffd6e7" },
  { id: 1,  seed: "Mia",   style: A, bg: "f9c6d0" },
  { id: 2,  seed: "Zoe",   style: A, bg: "e8d5f5" },
  { id: 3,  seed: "Luna",  style: A, bg: "d5e8f5" },
  { id: 4,  seed: "Aria",  style: A, bg: "ffecd2" },
  { id: 5,  seed: "Nova",  style: A, bg: "d5f5e8" },
  { id: 6,  seed: "Bella", style: A, bg: "f5e6d5" },
  { id: 7,  seed: "Ruby",  style: A, bg: "fde68a" },
  { id: 8,  seed: "Leo",   style: B, bg: "c0d8f5" },
  { id: 9,  seed: "Max",   style: B, bg: "bbf7d0" },
  { id: 10, seed: "Kai",   style: B, bg: "bfdbfe" },
  { id: 11, seed: "Finn",  style: B, bg: "ddd6fe" },
  { id: 12, seed: "Axel",  style: B, bg: "a7f3d0" },
  { id: 13, seed: "Cruz",  style: B, bg: "fed7aa" },
  { id: 14, seed: "Rex",   style: B, bg: "fca5a5" },
  { id: 15, seed: "Orion", style: B, bg: "d1fae5" },
];

const BASE = "https://api.dicebear.com/7.x";

export function avatarUrl(idOrDef: number | string | AvatarDef | undefined | null, size = 80): string {
  let def: AvatarDef | undefined;
  if (typeof idOrDef === "object" && idOrDef !== null && "seed" in idOrDef) {
    def = idOrDef as AvatarDef;
  } else {
    const num = Number(idOrDef ?? 0);
    def = AVATARS.find((a) => a.id === num) ?? AVATARS[0];
  }
  return `${BASE}/${def.style}/svg?seed=${def.seed}&backgroundColor=${def.bg}&size=${size}`;
}

export function getAvatar(id: number | string | undefined | null): AvatarDef {
  const num = Number(id ?? 0);
  return AVATARS.find((a) => a.id === num) ?? AVATARS[0];
}

File: ventigo_app/lib/services/avatars.dart
/// 16 hand-picked DiceBear avatars - direct port from avatars.ts.
class AvatarDef {
  final int id;
  final String seed;
  final String style; // "adventurer" | "adventurer-neutral"
  final String bg; // hex without #

  const AvatarDef({
    required this.id,
    required this.seed,
    required this.style,
    required this.bg,
  });
}

const _a = 'adventurer';
const _b = 'adventurer-neutral';

const List<AvatarDef> avatars = [
  AvatarDef(id: 0, seed: 'Lily', style: _a, bg: 'ffd6e7'),
  AvatarDef(id: 1, seed: 'Mia', style: _a, bg: 'f9c6d0'),
  AvatarDef(id: 2, seed: 'Zoe', style: _a, bg: 'e8d5f5'),
  AvatarDef(id: 3, seed: 'Luna', style: _a, bg: 'd5e8f5'),
  AvatarDef(id: 4, seed: 'Aria', style: _a, bg: 'ffecd2'),
  AvatarDef(id: 5, seed: 'Nova', style: _a, bg: 'd5f5e8'),
  AvatarDef(id: 6, seed: 'Bella', style: _a, bg: 'f5e6d5'),
  AvatarDef(id: 7, seed: 'Ruby', style: _a, bg: 'fde68a'),
  AvatarDef(id: 8, seed: 'Leo', style: _b, bg: 'c0d8f5'),
  AvatarDef(id: 9, seed: 'Max', style: _b, bg: 'bbf7d0'),
  AvatarDef(id: 10, seed: 'Kai', style: _b, bg: 'bfdbfe'),
  AvatarDef(id: 11, seed: 'Finn', style: _b, bg: 'ddd6fe'),
  AvatarDef(id: 12, seed: 'Axel', style: _b, bg: 'a7f3d0'),
  AvatarDef(id: 13, seed: 'Cruz', style: _b, bg: 'fed7aa'),
  AvatarDef(id: 14, seed: 'Rex', style: _b, bg: 'fca5a5'),
  AvatarDef(id: 15, seed: 'Orion', style: _b, bg: 'd1fae5'),
];

const _base = 'https://api.dicebear.com/7.x';

/// Build a DiceBear avatar URL. Uses PNG format for simpler caching.
String avatarUrl(dynamic idOrDef, {int size = 80}) {
  AvatarDef def;
  if (idOrDef is AvatarDef) {
    def = idOrDef;
  } else {
    final num = idOrDef is int ? idOrDef : int.tryParse(idOrDef?.toString() ?? '') ?? 0;
    def = getAvatar(num);
  }
  return '$_base/${def.style}/png?seed=${def.seed}&backgroundColor=${def.bg}&size=$size';
}

/// Get an AvatarDef by id, falling back to avatars[0].
AvatarDef getAvatar(int id) {
  return avatars.where((a) => a.id == id).firstOrNull ?? avatars[0];
}
